safenode/src/protocol/types/transfers/mod.rs

```diff
@@ -131,6 +155,8 @@ fn transfer(send_inputs: Inputs) -> Result<Outputs> {
 
     let mut change_dbc = None;
     created_dbcs.retain(|created| {
+        // This is not 100% solid in my mind, since we could have another output that is sending to ourselves.
+        // In the wallet (which calls this code) such an operation is skipped. But worth noting anyway.
         if created.dbc.public_address() == &change_to && change.as_nano() > 0 {
             change_dbc = Some(created.dbc.clone());
             false
```

Comment:
If any other DBCs are for us, we will remove them
```
             change_dbc = Some(created.dbc.clone());
             false
```
An overwrite the `change_dbc` variable. 
Effectively removing the DBC from both `created_dbcs` and `change_dbc`.